File: src/main/java/com/larryhsiao/clotho/color/RedInt.java
package com.larryhsiao.clotho.color;

import com.larryhsiao.clotho.Source;

/**
 * Extract blue int in color int
 */
public class RedInt implements Source<Long> {
    private final long color;

    public RedInt(long color) {this.color = color;}

    @Override
    public Long value() {
        return (color >> 16) & 0xff;
    }
}

File: src/main/java/com/larryhsiao/clotho/color/RedInt.kt
package com.larryhsiao.clotho.color

import com.larryhsiao.clotho.Source

/**
 * Extract red int in color int
 */
class RedInt(private val color: Long) : Source<Long> {
    override fun value(): Long {
        return color.shr(16) and 0xff
    }
}